### PR TITLE
Database task

### PR DESCRIPTION
**Pull request by tomrist9**
fix: Remove manual ID setting and implement auto-increment for User and Ticket

- Updated `saveUser` and `saveTicket` methods to exclude manual ID setting
- Added logic to retrieve auto-generated IDs using `RETURN_GENERATED_KEYS`
- Adjusted `Main` class to rely on database-generated IDs
- Ensured database schema supports auto-increment for ID columns
- Improved consistency and resolved primary key conflicts

### CODE REVIEW

File: andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class TicketDaoImpl implements TicketDao {
+
+
+    @Override
+    public void saveTicket(Ticket ticket) {
+        String query = "INSERT INTO Tickets (userId, ticketType, description) VALUES(?,?,?)";
+        try (Connection con = DatabaseConnection.getConnection(); PreparedStatement pstmt = con.prepareStatement(query)) {
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java]: formatting
after ; start from new line

File: andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public List<Ticket> getTicketsByUserId(int userId) {
+        String query = "SELECT * FROM ticket WHERE user_id=?";
+        List<Ticket> tickets = new ArrayList<>();
+        try (Connection con = DatabaseConnection.getConnection(); PreparedStatement pstm = con.prepareStatement(query)) {
+            pstm.setInt(1, userId);
+            ResultSet resultSet = pstm.executeQuery();
+            while (resultSet.next()) {
+                tickets.add(new Ticket(resultSet.getInt("id"), resultSet.getInt("user_id"), (TicketType) resultSet.getObject("ticket_type"), resultSet.getString("description")));
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java]: You could add method for create Ticket from ResultSet to separate Mapper class

**Review by Darya-Drutsko — COMMENTED** [review]
Good job!
But add also sql script for creating tables

File: andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java (hunk 1 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/example/dao/impl/TicketDaoImpl.java]: done

**Review by Darya-Drutsko — APPROVED** [review]

**Pull request merged**